operator/operator.go

```diff
@@ -160,6 +161,75 @@ func (operator *Operator) handleBlocks() error {
 	return nil
 }
 
+func (operator *Operator) handleChallenges() error {
+	challenges := make(chan *bindings.ErdstallChallenged)
+	sub, err := operator.contract.WatchChallenged(nil, challenges, nil, nil)
+	if err != nil {
+		return fmt.Errorf("creating challenge subcription: %w", err)
+	}
+	defer sub.Unsubscribe()
+
+	for {
+		select {
+		case err := <-sub.Err():
+			return fmt.Errorf("subscription error: %w", err)
+
+		case c := <-challenges:
+			log.Warnf("Operator.handleChallenges: Incoming challenge for account %v and epoch %v", c.Account.String(), c.Epoch)
+
+			tr, err := func() (*bind.TransactOpts, error) {
+				ctx, cancel := createDefaultContext()
+				defer cancel()
+
+				return operator.ethClient.NewTransactor(ctx, big.NewInt(0), eth.DefaultGasLimit, operator.ethClient.Account())
+			}()
```

Comment:
Don't you return a transactor here whose context is already closed when it's used?

Reply:
I checked the implementation of `NewTransactor`. It only uses the context during creation of the transactor (to retrieve nonce and gas price). The created transactor doesn't rely on the context afterwards.

Reply:
I see, also checked it. Then we have another problem: our`TransactOpts` never have any context set :joy: so there's no timeout on any blockchain requests that we do with those transactors.

Reply:
So best solution now is to move the single challenge handling into its own method, as proposed above, and then use the same context for creating the transactor as well as setting it as the `Ctx` field, so that below `Exit` call has the context set. Putting it into its own method ensures that the `defer cancel()` calls don't accumulate in the loop and would only be called all at once when this loop stops (i.e., at the end of the routine/program).

Reply:
ok, done